datafusion/proto/src/from_proto.rs

```diff
@@ -806,11 +806,20 @@ pub fn parse_expr(
                         .ok_or_else(|| Error::unknown("BuiltInWindowFunction", *i))?
                         .into();
 
+                    let args = match parse_optional_expr(&expr.expr, registry)? {
+                        None => {
+                            vec![]
+                        }
+                        Some(x) => {
```

Comment:
is there a better approach  ? 🤔

Reply:
How about something like (untested):

```rust
let args = parse_optional_expr(&expr.expr, registry)?
  .map(Vec::new)
  .unwrap_or_else(|| vec![]);
```

Not sure if that is better but it is more functional

Reply:
souds good, i try it but seems `Vec::new` can not create one size vec with one agrs.

Reply:
Maybe 

```rust
let args = parse_optional_expr(&expr.expr, registry)?
  .map(|e| vec![e])
  .unwrap_or_else(|| vec![]);
```

Reply:
ye 😂,  thanks